### DISCUSSION TITLE
v1.5.0

### DISCUSSION THREAD

**Discussion by halbmy** [gimli-org/pyGIMLi#668] [Show and tell]
## What changed
* Re-enable 3D images on website, now including interactive mode using pyvista's `DynamicScraper`
* Fix iteration number in pre/postStep by @prisae in https://github.com/gimli-org/gimli/pull/643
* Typo correction by @soehag in https://github.com/gimli-org/gimli/pull/650
* FIX adding custom constrains matrix into lazy evaluation process of regularization settings (fixes #659)
* Enable `DataContainer.subset()` for geometric filtering (e.g. extract a line)
* Reciprocal data analysis: `ert.reciprocalAnalysis(data)`, `ert.fitReciprocalModel(data)`, `ert.reciprocalProcessing(data)`
* Extend reciprocal analysis to `TimelapseERT` class
* Improved display of ERT data: using, x/y fields or predefined styles (e.g. A-MN)
* Improved maxtrix `pg.show(mat)` for `Matrix`, `RealNumpyMatrix`, `IdentityMatrix`, `DiagonalMatrix`, all others
* Improve docstrings for several classes and functions
* Extend setting mesh properties `mesh[str]=val`, including markers
* New example on tree tomography using ERT, styling other examples
* Improve details in several examples and correct inconsistencies
* Add new papers using pyGIMLi
* Fix several bugs: res2dinv import, bug with scipy=1.12, 
* Updated to new version of pydata-sphinx-theme

## New Contributors
* @soehag made their first contribution in https://github.com/gimli-org/gimli/pull/650

**Full Changelog**: https://github.com/gimli-org/gimli/compare/v1.4.6...v1.5.0

<hr /><em>This discussion was created from the release <a href='https://github.com/gimli-org/gimli/releases/tag/v1.5.0'>v1.5.0</a>.</em>

**Comment by florian-wagner** [comment]
Finally interactive 3D Plots on the website (e.g. https://www.pygimli.org/_examples_auto/5_flow_transport/plot_3D_Darcy_flow.html) due to the `DynamicScraper` in [PyVista](https://www.pyvista.org) by @banesullivan and Co.!

**Reply by banesullivan** [reply]
AAAhhhhh!!! So Awesome!!! 
![sphx_glr_plot_3D_Darcy_flow_002](https://github.com/gimli-org/gimli/assets/22067021/364b82c7-3694-43aa-8d52-828c7eba7f7d)

**Reply by halbmy** [reply]
Thanks to you and your great work! This will definitely boost solving 3D problems (ERT, gravity, magnetics, but also EM) with pyGIMLi.
